internal/operator/client.go

```diff
@@ -104,12 +111,21 @@ func NewDynamicClient() (dynamic.Interface, error) {
 	return dynamic.NewForConfig(kubeConfig())
 }
 
-// NewKubernetesClient returns a kubernetes clientset
-func NewKubernetesClient() (*kubernetes.Clientset, error) {
-	return kubernetes.NewForConfig(kubeConfig())
-}
-
 func NewConfigClient() (*configv1.ConfigV1Client, error) {
 	// create openshift config clientset
 	return configv1.NewForConfig(kubeConfig())
 }
+
+func NewFakeClient() operatorClient {
+	scheme := runtime.NewScheme()
+	operatorv1alpha1.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+	operatorv1.AddToScheme(scheme)
+	pkgserverv1.AddToScheme(scheme)
+	apiextensionsv1.AddToScheme(scheme)
+	client := fake.NewClientBuilder().WithScheme(scheme).Build()
+	var operatorClient operatorClient = operatorClient{
+		Client: client,
+	}
+	return operatorClient
+}
```

Comment:
I think we could have one fake client per test with only the resources that particular client needs. And also feel like this belongs to the test file not in here.

Reply:
Thanks for the review Alex, Made the changes with the new commit.